src/tree-node/node-label.js

```diff
@@ -49,7 +49,7 @@ class NodeLabel extends PureComponent {
     }
 
     return (
-      <label title={title || label} htmlFor={id}>
+      <label title={title || label}>
         <Checkbox
           name={id}
```

Comment:
Now I'm thinking, the `name={id}` is not unique in the page when you have multiple instances. This is bad HTML semantically. Had these been radio buttons, the problem would have been more severe. 

I added https://github.com/dowjones/react-dropdown-tree-select/issues/215 to keep track of it.